src/core/network/createNetwork.js

```diff
@@ -29,33 +29,44 @@ export default (config, logger, lifecycle, networkStrategy) => {
      * @property {function} send - call this function when you are ready to send the payload
      * @property {Object} payload - payload object that will be sent
      * @property {Promise} responsePromise - promise that will yield the raw response body
-     * @property {boolean} isBeacon - whether or not this request is a beacon request
      */
     /**
      * Create a new request.  Once "send" on the returned object is called, the lifecycle
-     *  method "onBeforeSend" will be triggered with { payload, responsePromise, isBeacon } as
+     *  method "onBeforeSend" will be triggered with { payload, responsePromise } as
      *  the parameter.  When the response is returned it will call the lifecycle method "onResponse"
      *  with the returned response object.
-     * @param {boolean} isBeacon - true to send a beacon (defaults to false).  If you send
-     *   a beacon, no data will be returned.
-     *
+     * @param {boolean} [expectsResponse=true] The endpoint and request mechanism
+     * will be determined by whether a response is expected.
      * @returns {Request}
      */
-    newRequest(isBeacon = false) {
+    newRequest(expectsResponse = true) {
       const payload = createPayload();
-      const action = isBeacon ? "collect" : "interact";
+      const action = expectsResponse ? "interact" : "collect";
       const url = `https://${edgeDomain}/${action}?propertyID=${propertyID}`;
       const deferred = defer();
       const responsePromise = deferred.promise
         .then(() =>
-          lifecycle.onBeforeSend({ payload, responsePromise, isBeacon })
+          lifecycle.onBeforeSend({
```

Comment:
It's possible for a component to handle the onBeforeSend lifecycle method, and add an event with a query even though this request is a beacon.  We could add a guard for this in the payload object's addEvent method.

Reply:
It's a good point. Let's talk about this in today's check-in. We may actually want components to be able to add events with a query from within `onBeforeSend`. If so, we have an order-of-operations problem because we're calculating whether a response is expected before `onBeforeSend` is called. I think we need to come to an agreement about the purpose of `onBeforeSend` and the rules about what you can/can't do within that lifecycle phase.